earth_enterprise/src/fusion/gepublish/CurlRequest.h

```diff
@@ -26,7 +26,106 @@
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, GetRequestWriteFunc) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, UploadRequestReadFunc) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, std::nullptr_t) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, void*) = 0;
+  virtual curl_slist*slist_append(curl_slist* lst, const char * item) = 0;
+  virtual void slist_free_all(curl_slist* lst) = 0;
+  virtual void easy_cleanup(CURL *handle) = 0;
+  virtual CURLcode easy_perform(CURL * handle) = 0;
+  virtual CURLcode easy_getinfo(CURL *curl, CURLINFO info, long* result_code) = 0;
+};
+
+namespace {
+struct curl_api_impl : public curl_api {
+  CURL* easy_init() override
+  {
+    return curl_easy_init();
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, bool param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, int param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, long param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, unsigned long param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, const char* param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, curl_slist* param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, CurlRequestWriteFunc param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, GetRequestWriteFunc param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, UploadRequestReadFunc param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, void* param) override
+  {
+    return curl_easy_setopt(handle, option, param);
+  }
+  CURLcode easy_setopt(CURL *handle, CURLoption option, std::nullptr_t) override
+  {
+    return curl_easy_setopt(handle, option, nullptr);
+  }
+  curl_slist*slist_append(curl_slist* lst, const char * item) override
+  {
+    return curl_slist_append(lst, item);
+  }
+  void slist_free_all(curl_slist* lst) override
+  {
+    curl_slist_free_all(lst);
+  }
+  void easy_cleanup(CURL *handle) override
+  {
+    curl_easy_cleanup(handle);
+  }
+  CURLcode easy_perform(CURL * handle) override
+  {
+    return curl_easy_perform(handle);
+  }
+  CURLcode easy_getinfo(CURL *handle, CURLINFO info, long* result_code) override
+  {
+    return curl_easy_getinfo(handle, info, result_code);
+  }
+} the_curl_api_impl;
```

Comment:
This is the real implementation that will be used when this code isn't under test.